tests/tests_geomstats/test_spd_matrices.py

```diff
@@ -19,192 +22,156 @@
     SPDMetricEuclidean,
     SPDMetricLogEuclidean,
 )
+from tests.conftest import Parametrizer, generate_tests
 
 SQRT_2 = math.sqrt(2)
 
 
-class TestSPDMatrices(geomstats.tests.TestCase):
-    """Test of SPDMatrices methods."""
-
-    def setup_method(self):
-        """Set up the test."""
-        warnings.simplefilter("ignore", category=ImportWarning)
+def belongs_data():
+    smoke_data = [
+        dict(n=2, mat=[[3.0, -1.0], [-1.0, 3.0]], expected=True),
+        dict(n=2, mat=[[1.0, 1.0], [2.0, 1.0]], expected=False),
+        dict(
+            n=3,
+            mat=[[1.0, 2.0, 3.0], [2.0, 4.0, 5.0], [3.0, 5.0, 6.0]],
+            expected=True,
+        ),
+        dict(
+            n=2,
+            mat=[[[1.0, 0.0], [0.0, 1.0]], [[1.0, -1.0], [0.0, 1.0]]],
+            expected=[True, False],
+        ),
+    ]
+    return generate_tests(smoke_data)
+
+
+def random_point_data():
+    smoke_data = [
+        dict(n=1, num_points=1),
+        dict(n=2, num_points=1),
+        dict(n=10, num_points=10),
+        dict(n=10, num_points=1000),
+    ]
+    return generate_tests(smoke_data)
+
+
+def logm_data():
+    smoke_data = [
+        dict(spd_mat=[[1.0, 0.0], [0.0, 1.0]], expected=[[0.0, 0.0], [0.0, 0.0]])
+    ]
+    return generate_tests(smoke_data)
+
+
+def cholesky_factor_data():
+    smoke_data = [
+        dict(
+            spd_mat=[[2.0, 0.0, 0.0], [0.0, 2.0, 0.0], [0.0, 0.0, 2.0]],
+            cf=[[SQRT_2, 0.0, 0.0], [0.0, SQRT_2, 0.0], [0.0, 0.0, SQRT_2]],
+        )
+    ]
+    return generate_tests(smoke_data)
 
-        gs.random.seed(1234)
 
-        self.n = 3
-        self.space = SPDMatrices(n=self.n)
-        self.metric_affine = SPDMetricAffine(n=self.n)
-        self.metric_bureswasserstein = SPDMetricBuresWasserstein(n=self.n)
-        self.metric_euclidean = SPDMetricEuclidean(n=self.n)
-        self.metric_logeuclidean = SPDMetricLogEuclidean(n=self.n)
-        self.n_samples = 4
+def differential_cholesky_factor_data(tangent_vec, base_point, expected):
+    smoke_data = [
+        dict(
+            tangent_vec=[[1.0, 1.0], [1.0, 1.0]],
+            base_point=[[4.0, 2.0], [2.0, 5.0]],
+            expected=[[1 / 4, 0.0], [3 / 8, 1 / 16]],
+        )
+    ]
+    return generate_tests(smoke_data)
+
+
+def differential_power_data(power, tangent_vec, base_point, expected):
+    smoke_data = [
+        dict(
+            power=0.5,
+            tangent_vec=[[2.0, 1.0, 1.0], [1.0, 0.5, 0.5], [1.0, 0.5, 0.5]],
+            base_point=[[1.0, 0.0, 0.0], [0.0, 2.5, 1.5], [0.0, 1.5, 2.5]],
+            expected=[
+                [1.0, 1 / 3, 1 / 3],
+                [1 / 3, 0.125, 0.125],
+                [1 / 3, 0.125, 0.125],
+            ],
+        )
+    ]
+    return generate_tests(smoke_data)
+
+
+def inverse_differential_power_data(power, tangent_vec, base_point, expected):
```

Comment:
Should this function take input args? They are not used + upon calling the function, no args is passed as input in the test files.